lib/Mongo/MongoClient.php

```diff
@@ -115,6 +117,12 @@ public function close($connection = null)
      */
     public function connect()
     {
+        try {
+            $this->forceConnect();
+        } catch (\MongoDB\Driver\Exception\Exception $e) {
+            ExceptionConverter::toLegacy($e);
+        }
```

Comment:
This is one of those cases where I'd just opt for the BC break and use lazy connections.